nxc/modules/snipped.py

```diff
@@ -0,0 +1,120 @@
+        """
+        USERS           Download only specified user(s); format: -o USERS=user1,user2,user3
+        """
+        self.context = context
+        self.screenshot_path_stub = r"Pictures\Screenshots"
+        self.users = module_options["USERS"].split(",") if "USERS" in module_options else None
+
+    def on_admin_login(self, context, connection):
+        self.context = context
+        self.connection = connection
+        self.share = "C$"
+        
+        host = f"{connection.hostname}.{connection.domain}"
+        domain = connection.domain
+        username = connection.username
+        kerberos = connection.kerberos
+        aesKey = connection.aesKey
+        use_kcache = getattr(connection, "use_kcache", False)
+        password = getattr(connection, "password", "")
+        lmhash = getattr(connection, "lmhash", "")
+        nthash = getattr(connection, "nthash", "")
+
+        target = Target.create(
+            domain=domain,
+            username=username,
+            password=password,
+            target=host,
+            lmhash=lmhash,
+            nthash=nthash,
+            do_kerberos=kerberos,
+            aesKey=aesKey,
+            use_kcache=use_kcache,
+        )
+
+        dploot_conn = self.upgrade_connection(target=target, connection=connection.conn)
+
+        output_path = f"nxc_snipped_{connection.host}"
+        context.log.debug("Getting all user folders")
+        try:
+            user_folders = dploot_conn.listPath(self.share, "\\Users\\*")
+        except Exception as e:
+            context.log.fail(f"Failed to list user folders: {e}")
+            return
+
+        context.log.debug(f"User folders: {user_folders}")
+        if not user_folders:
+            context.log.fail("No User folders found!")
+            return
+        else:
+            context.log.display("Attempting to download screenshots if existent.")
+
+        for user_folder in user_folders:
+            if not user_folder.is_directory():
+                continue
+            folder_name = user_folder.get_longname()
+            if folder_name in [".", "..", "All Users", "Default", "Default User", "Public"]:
+                continue
+            if self.users and folder_name not in self.users:
+                continue
+
+            screenshot_path = ntpath.normpath(join(r"Users", folder_name, self.screenshot_path_stub))
+            try:
+                screenshot_files = dploot_conn.listPath(self.share, screenshot_path + "\\*")
+            except Exception as e:
+                context.log.debug(f"Screenshot folder {screenshot_path} not found for user {folder_name}: {e}")
+                continue
+
+            if not screenshot_files:
+                context.log.debug(f"No screenshots found in {screenshot_path} for user {folder_name}")
+                continue
+
+            user_output_dir = join(output_path, folder_name)
+            if not exists(user_output_dir):
+                makedirs(user_output_dir)
```

Comment:
Please save downloaded files from the module in `{NXC_PATH}/modules/MODULENAME/*` similar to this:
https://github.com/Pennyw0rth/NetExec/blob/78da498846da4419f937cdde81a017164b932b4e/nxc/modules/pre2k.py#L70-L72

Reply:
https://github.com/Yeeb1/NetExec/commit/969969b646354f25c3340fb65bdb5665e4d5dde6#diff-f33ecc9861a187049943d18be14e4a7ffcadbf3a32fd5e3dae7773017e020e2cR60

Modified output directory similar to pre2k module